### PR TITLE
Replaces /atom.initialized and .admin_spawned with flags

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
As `initialized` is guaranteed to be written to, this gives us a VERY minor speed boost when reading `flags_1`. This might be wholly offset by the cost added by the bitwise operation for things that touch `initialized`, probably a wash. Other than that it'll save a trifling amount of memory if flags_1 would be written to anyways (probably true for items?)

@MrStonedOne might want to give this a look-over to see if I haven't missed anything, this is one of those 3 AM ideas

:cl: Naksu
code: Replaced initialized and admin_spawned vars with flags.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
While cleaning up vars on atoms is noble, I do still kinda wonder what's with the obsession with making everything into a flag as opposed to a var? As far as I know, manipulating a var is cheaper than dealing with bitflags

On that note, why bother with bitflags are all? having `flags_1` and `flags_2`; if performance is being discarded, why not just have a single `misc_attributes` list and just have text based flags for them? Much like species traits were made into? Ie: `if(INITIALIZED in atom_traits)` or something like that.

**Comment by MrStonedOne** [comment]
bitwise math is a bit slower then a direct var access, yes, but nowhere near the level of abusing `in` on lists.

The main reason is var access speed is correlated with the number of changed(from compile time) vars the object has. The more changed variables on an object, the higher the var access cost is for that object. 

There is also a minor memory component, moving all of these flags to vars would likely add 25 to 35mb to our memory usage.  This is less of a reason why and more of a added bonus.

**Comment by MrStonedOne** [comment]
```dm
[23:03:43] <%MrStonedOne> !dm /datum/test/var/flag = 17;;;var/datum/test/T = new();var/temp = 0;;BENCHM("direct", 2, temp = (T.flag ? 1 : 0));;BENCHM("bitwise", 2, temp = (T.flag & 16));;
[23:03:45] <+Bot32> TheAlmightyStoner: direct: 4 ds;  bitwise: 3 ds
```

bitwise might actually be faster 0-0

**Review by MrsTonedOne — APPROVED** [review]

File: code/game/machinery/droneDispenser.dm [hunk 1]
```diff
@@ -174,7 +174,7 @@
 				use_power(power_used)
 
 			var/atom/A = new dispense_type(loc)
-			A.admin_spawned = admin_spawned
+			A.flags_1 |= (flags_1 | ADMIN_SPAWNED_1)
```
Comment (SpaceManiac) [inline code/game/machinery/droneDispenser.dm]: Incorrect logic here, should be more like
```dm
A.flags_1 = (A.flags_1 & ~ADMIN_SPAWNED_1) | (flags_1 & ADMIN_SPAWNED_1)
```
or so. Ditto in `drones_as_items.dm`, `hivelord.dm`, and `spawner.dm`.

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -2059,7 +2059,7 @@
 					else
 						var/atom/O = new path(target)
 						if(O)
-							O.admin_spawned = TRUE
+							O.flags_1 = ADMIN_SPAWNED_1
```
Comment (SpaceManiac) [inline code/modules/admin/topic.dm]: Should be `|=`.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/game/machinery/droneDispenser.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/game/machinery/droneDispenser.dm]: This isn't intended to clear the admin spawned flag if it's set on the new atom, though it is guaranteed to not be set on any new atom. "admin spawned" is, as far as i understand it, intended to be a viral property that taints an atom and any atom it creates.

File: code/modules/admin/topic.dm (hunk 2 above)
Comment (vuonojenmustaturska) [reply code/modules/admin/topic.dm]: Fixed. Thanks.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**